extensions/amp-ad/0.1/amp-ad-api-handler.js

```diff
@@ -169,16 +170,16 @@ export class AmpAdApiHandler {
 
   /** @override  */
   unlayoutCallback() {
-    this.unlisteners_.forEach(unlistener => unlistener());
-    this.unlisteners_ = [];
     if (this.iframe_) {
+      this.unlisteners_.forEach(unlistener => unlistener());
+      this.unlisteners_.length = 0;
+      this.embedStateApi_.destroy();
+      this.embedStateApi_ = null;
+      this.intersectionObserver_.destroy();
```

Comment:
That's a different `this`, but we should be destroying when there's no content. I'll add that.